modules/rational-compile.el

```diff
@@ -99,12 +130,58 @@ D could be a single directory or a list of directories."
             (expand-file-name f user-emacs-directory))
           rational-compile-init-files-list))
 
-;; A function to execute the compilation of the modules directory:
+(defun rational-compile--config-files-list ()
+  "Returns a list of the init files."
+  (mapcar (lambda (f)
+            (expand-file-name f rational-config-path))
+          rational-compile-config-files-list))
+;;(rational-compile--config-files-list)
+
+;; A function to execute the compilation of the modules within the list:
 (defun rational-compile-modules ()
-  "Compile (native or byte-code) the modules' source code for faster startups."
+  "Compile specific files defined in `rational-compile-module-list'.
+
+If `rational-compile-init-files' is non-nil, then it compiles
+the init-files (`rational-compile-init-files-list') within
+`user-emacs-directory'.
+
+If `rational-compile-user-configuration' is non-nil, then it
+compiles the config-files (`rational-compile-config-files-list')
+within `rational-config-path'.
+
+If `rational-compile-modules' is non-nil, then it compiles all
+the source-files for the modules defined in
+`rational-compile-module-list' within the subdirectory 'modules'
+in `user-emacs-directory', or in any directory specified in
+`rational-compile-extra-directories-list'.
+
+If `rational-compile-user-modules' is non-nil, then it compiles
+all the source-files for the modules defined in
+`rational-compile-module-list' within the subdirectory 'modules'
+in `rational-config-path'.
+
+If any source-file for any module specified in
+`rational-compile-module-list' doesn't exist within the paths
+specified, then it is ignored without a warning."
   (interactive)
-  (rational-compile-directory
-   (rational-compile--config-dirs-list)))
+  (when rational-compile-init-files
+    (rational-compile-init))
+  (when rational-compile-user-configuration
+    (rational-compile-config))
+  (when rational-compile-modules
+    (dolist (module rational-compile-module-list)
+      (let ((module-src (rational-locate-module-file module
+                                                     (flatten-list
+                                                      `(,(expand-file-name "modules/" user-emacs-directory)
+                                                        ,rational-compile-extra-directories-list))))
+        (when module-src
+          (rational-compile-file module-src))))))
+  (when rational-compile-user-modules
+    (dolist (module rational-compile-module-list)
+      (let ((module-src (rational-locate-module-file module
+                                                     `(,(expand-file-name "modules/" rational-config-path)))))
```

Comment:
After #136, this should be `custom-modules`. If we prefer just modules, will need an update to `init.el` as well to handle that name change.